web/packages/teleport/src/Roles/RoleEditor/RoleEditorAdapter.tsx

```diff
@@ -0,0 +1,168 @@
+import { ButtonSecondary } from 'design/Button';
+import Image from 'design/Image';
+
+import { State as ResourcesState } from 'teleport/components/useResources';
+import { Role, RoleWithYaml } from 'teleport/services/resources';
+import { yamlService } from 'teleport/services/yaml';
+import { YamlSupportedResourceKind } from 'teleport/services/yaml/types';
+import { ButtonLockedFeature } from 'teleport/components/ButtonLockedFeature';
+
+import { RoleEditor } from './RoleEditor';
+import tagpromo from './tagpromo.png';
+
+/**
+ * This component is responsible for converting from the `Resource`
+ * representation of a role to a more accurate `RoleWithYaml` structure. The
+ * conversion is asynchronous and it's performed on the server side.
+ */
+export function RoleEditorAdapter({
+  resources,
+  onSave,
+  onDelete,
+  onCancel,
+}: {
+  resources: ResourcesState;
+  onSave: (role: Partial<RoleWithYaml>) => Promise<void>;
+  onDelete: () => Promise<void>;
+  onCancel: () => void;
+}) {
+  const theme = useTheme();
+  const [convertAttempt, convertToRole] = useAsync(
+    async (yaml: string): Promise<RoleWithYaml | null> => {
+      if (resources.status === 'creating' || !resources.item) {
+        return null;
+      }
+      return {
+        yaml,
+        object: await yamlService.parse<Role>(YamlSupportedResourceKind.Role, {
+          yaml,
+        }),
+      };
+    }
+  );
+
+  const originalContent = resources.item?.content ?? '';
+  useEffect(() => {
+    convertToRole(originalContent);
+  }, [originalContent]);
+
+  return (
+    <Flex flex="1">
+      <Flex
+        flexDirection="column"
+        borderLeft={1}
+        borderColor={theme.colors.interactive.tonal.neutral[0]}
+        backgroundColor={theme.colors.levels.surface}
+        width="700px"
+      >
+        {convertAttempt.status === 'processing' && (
+          <Flex
+            flexDirection="column"
+            alignItems="center"
+            justifyContent="center"
+            flex="1"
+          >
+            <Indicator />
+          </Flex>
+        )}
+        {convertAttempt.status === 'error' && (
+          <Danger>{convertAttempt.statusText}</Danger>
+        )}
+        {convertAttempt.status === 'success' && (
+          <RoleEditor
+            originalRole={convertAttempt.data}
+            onCancel={onCancel}
+            onSave={onSave}
+            onDelete={onDelete}
+          />
+        )}
+      </Flex>
+      <Flex flex="1" alignItems="center" justifyContent="center">
+        <Box>
+          <H1 mb={2}>Teleport Policy</H1>
+          <Flex mb={4}>
+            <Box
+              flex="1"
+              css={`
+                width: min-content;
+              `}
+            >
+              <P>
+                TAG serves as a powerful tool for both technical and security
+                teams to maintain robust access control structures, ensuring
+                security, compliance, and operational efficiency.
+              </P>
```

Comment:
Isn't the name just `Teleport Policy` now and not TAG? Also, if it is still TAG, should we instead use the full name `Teleport Access Graph` for those not in the know?

Reply:
Oh, yes, @zmb3 already pointed it out. The copy turned out to be outdated. I'll update it.